internal/db/query_execution.go

```diff
@@ -191,6 +203,7 @@ func (qb Q) performJoinOne(tx *holdTx, outer []*QueryResult, j JoinOperation) []
 				outer[i].Empty()
 			}
 		}
+		qb.projectFields(qb.Selections[j.To.AliasID], outer)
```

Comment:
the fact that we need to call projectFields in so many places suggests a refactor is required here, but it's fine to punt on that for now :)

Reply:
:)